tests/dhcp_server/test_dhcp_server.py

```diff
@@ -0,0 +1,336 @@
+    },
+    {
+        'description': 'Verify configured interface with \
+                        client mac in FDB table can successfully get IP',
+        'dut_port_to_capture_pkt': 'Ethernet1',
+        'expected_assigned_ip': '192.168.0.11',
+        'ptf_port_index': 1,
+        'refresh_fdb_ptf_port': 'eth1',
+        "test_xid": 12
+    },
+    {
+        'description': 'Verify configured interface with \
+                        client mac in FDB table but mac was learnt \
+                        from another interface can successfully get IP.',
+        'dut_port_to_capture_pkt': 'Ethernet2',
+        'expected_assigned_ip': '192.168.0.12',
+        'ptf_mac_port_index': 3,
+        'ptf_port_index': 2,
+        'refresh_fdb_ptf_port': 'eth3',
+        "test_xid": 13
+    },
+    {
+        'description': 'Verify no-configured interface cannot get IP',
+        'dut_port_to_capture_pkt': 'any',
+        "expected_assigned_ip": None,
+        'ptf_port_index': 4,
+        'refresh_fdb_ptf_port': 'eth4',
+        "test_xid": 14
+    }
+])
+def test_dhcp_server_port_based_assignment_single_ip(duthost, ptfhost, ptfadapter, senario_context):
+    """
+       Test single ip assignment with different senarios, each senario has a description in senario_context.
+    """
+    test_xid = senario_context['test_xid']
+    ptf_mac_port_index = senario_context.get('ptf_mac_port_index', senario_context['ptf_port_index'])
+    client_mac = ptfadapter.dataplane.get_mac(0, ptf_mac_port_index).decode('utf-8')
+    expected_ip = senario_context['expected_assigned_ip']
+    pkts_validator = validate_dhcp_server_pkts if expected_ip else validate_no_dhcp_server_pkts
+    pkts_validator_args = [test_xid, expected_ip, DHCP_MESSAGE_TYPE_OFFER_NUM] if expected_ip else [test_xid]
+    discover_pkt = create_dhcp_client_packet(
+        src_mac=client_mac,
+        message_type=DHCP_MESSAGE_TYPE_DISCOVER_NUM,
+        client_options=[],
+        xid=test_xid
+    )
+    send_and_verify(
+        duthost=duthost,
+        ptfhost=ptfhost,
+        ptfadapter=ptfadapter,
+        senario_context=senario_context,
+        test_pkt=discover_pkt,
+        pkts_validator=pkts_validator,
+        pkts_validator_args=pkts_validator_args
+    )
+    request_pkt = create_dhcp_client_packet(
+        src_mac=client_mac,
+        message_type=DHCP_MESSAGE_TYPE_REQUEST_NUM,
+        client_options=[("requested_addr", expected_ip), ("server_id", VLAN_GATE_WAY)],
+        xid=test_xid
+    )
+    pkts_validator_args = [test_xid, expected_ip, DHCP_MESSAGE_TYPE_ACK_NUM] if expected_ip else [test_xid]
+    send_and_verify(
+        duthost=duthost,
+        ptfhost=ptfhost,
+        ptfadapter=ptfadapter,
+        senario_context=senario_context,
+        test_pkt=request_pkt,
+        pkts_validator=pkts_validator,
+        pkts_validator_args=pkts_validator_args
+    )
+
+
+@pytest.mark.parametrize('senario_context', [
+    {
+        'dut_port_to_capture_pkt': 'Ethernet5',
```

Comment:
Same as above

Reply:
Yes, added code to parse VLAN and port relationship between DUT and PTF from running config instead of const.